src/main/java/com/example/webapp/Controller/DepartmentController.java

```diff
@@ -11,34 +13,35 @@
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 
-import javax.validation.Valid;
 import java.util.List;
 
 @RestController
 public class DepartmentController {
 
-        @Autowired
-        DepartmentService departmentService;
-
-        @GetMapping("/departments")
-        public ResponseEntity<List<Department>> getDepartments() {
-
-            return departmentService.getAll();
-        }
-
-        @GetMapping("/departments/{id}")
-        public ResponseEntity<Department> getDepartmentById(@PathVariable long id) {
-
-            return departmentService.getById(id);
-        }
-
-        @PostMapping("/departments")
-        public ResponseEntity<Department> addDepartment(@RequestBody Department department) {
-            return departmentService.add(department);
-        }
-
-        @PutMapping("/departments/{id}")
-    public ResponseEntity<Department> updateDepartment(@PathVariable long id, @RequestBody Department department){
-            return departmentService.updateDepartment(id, department);
-        }
-    }
\ No newline at end of file
+    @Autowired
+    DepartmentService departmentService;
+
+    @GetMapping("/departments")
+    public ResponseEntity<List<Department>> getDepartments() {
+        List<Department> list = departmentService.getAll();
+        return new ResponseEntity<>(list, HttpStatus.OK);
+    }
+
+    @GetMapping("/departments/{id}")
+    public ResponseEntity<Department> getDepartmentById(@PathVariable long id) {
+        Department byId = departmentService.getById(id);
+        return new ResponseEntity<>(byId, HttpStatus.OK);
```

Comment:
Would recommend to use factory methods of ResponseEntity, example ResponseEntity.ok(body)

Reply:
Done